client/state/memdb.go

```diff
@@ -28,15 +29,19 @@ type MemDB struct {
 	// drivermanager -> plugin-state
 	driverManagerPs *driverstate.PluginState
 
+	logger hclog.Logger
+
 	mu sync.RWMutex
 }
 
-func NewMemDB() *MemDB {
+func NewMemDB(logger hclog.Logger) *MemDB {
+	logger = logger.Named("memdb")
 	return &MemDB{
 		allocs:         make(map[string]*structs.Allocation),
 		deployStatus:   make(map[string]*structs.AllocDeploymentStatus),
 		localTaskState: make(map[string]map[string]*state.LocalState),
 		taskState:      make(map[string]map[string]*structs.TaskState),
+		logger:         logger,
```

Comment:
I temporarily used it to debug test failures and wanted to make it easier for future debuggers.

Reply:
👍  this seems nice for quality of life when writing annoying tests